packages/twenty-server/src/database/commands/upgrade-version-command/1-5/1-5-populate-message-folder-fields.command.ts

```diff
@@ -0,0 +1,322 @@
+import { InjectDataSource, InjectRepository } from '@nestjs/typeorm';
+
+import { Command } from 'nest-commander';
+import { DataSource, In, IsNull, Repository } from 'typeorm';
+
+import {
+  ActiveOrSuspendedWorkspacesMigrationCommandRunner,
+  type RunOnWorkspaceArgs,
+} from 'src/database/commands/command-runners/active-or-suspended-workspaces-migration.command-runner';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { WorkspaceMetadataVersionService } from 'src/engine/metadata-modules/workspace-metadata-version/services/workspace-metadata-version.service';
+import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
+import { WorkspaceSchemaManagerService } from 'src/engine/twenty-orm/workspace-schema-manager/workspace-schema-manager.service';
+import { MessageChannelWorkspaceEntity } from 'src/modules/messaging/common/standard-objects/message-channel.workspace-entity';
+import { MessageFolderWorkspaceEntity } from 'src/modules/messaging/common/standard-objects/message-folder.workspace-entity';
+import { SyncMessageFoldersService } from 'src/modules/messaging/folder-sync-manager/services/sync-message-folders.service';
+import { MessageFolderName } from 'src/modules/messaging/message-import-manager/drivers/microsoft/types/folders';
+
+type MessageFolder = Pick<
+  MessageFolderWorkspaceEntity,
+  'name' | 'isSynced' | 'isSentFolder' | 'externalId'
+>;
+
+@Command({
+  name: 'upgrade:1-5:populate-message-folder-fields',
+  description:
+    'Populate isSynced, isSentFolder, and externalId fields for existing message folders. ' +
+    'First populates basic fields for existing hardcoded folders, then runs sync service ' +
+    'to get complete folder data with correct external IDs from email providers.',
+})
+export class PopulateMessageFolderFieldsCommand extends ActiveOrSuspendedWorkspacesMigrationCommandRunner {
```

Comment:
@neo773 What's the behavior for existing messageChannel if we don't run this command? Will the sync break?

Running upgrade commands can take to ~3hours on production as we have a lot of workspaces (~3k)

Reply:
I wonder if this command is worth it, as we should sync folders on every sync anyway, it's seems overkill